src/main/java/net/javadiscord/javabot/systems/help/AutoCodeFormatter.java

```diff
@@ -0,0 +1,102 @@
+package net.javadiscord.javabot.systems.help;
+
+import lombok.AllArgsConstructor;
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.entities.MessageEmbed;
+import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
+import net.javadiscord.javabot.data.config.BotConfig;
+import net.javadiscord.javabot.systems.moderation.AutoMod;
+import net.javadiscord.javabot.systems.user_preferences.UserPreferenceService;
+import net.javadiscord.javabot.systems.user_preferences.model.Preference;
+import net.javadiscord.javabot.util.ExceptionLogger;
+import net.javadiscord.javabot.util.WebhookUtil;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import javax.annotation.Nonnull;
+import java.util.Objects;
+
+/**
+ * Handles auto-formatting of code in help-channels.
+ * A non-listener class since it has to be called strictly after {@link HelpListener} to not send its messages before channel reserving.
+ */
+@AllArgsConstructor
+public class AutoCodeFormatter {
+	private final AutoMod autoMod;
+	private final BotConfig botConfig;
+	private final UserPreferenceService preferenceService;
+
+	/**
+	 * Method responsible for finding a codeblock, if present.
+	 *
+	 * @param event a {@link MessageReceivedEvent}.
+	 * @return a MessageCodeblock instance, holding a startIndex, content and an endIndex
+	 */
+	@Nullable
+	private static AutoCodeFormatter.CodeBlock findCodeblock(@NotNull MessageReceivedEvent event) {
```

Comment:
This method does not really "find a codeblock" but finds a place where a code block should be inserted.